core/capabilities/launcher.go

```diff
@@ -0,0 +1,525 @@
+				// When this is solved, we can move to a generic aggregator
+				// and remove this.
+				triggerCap := remote.NewTriggerSubscriber(
+					cfg,
+					info,
+					*toDONInfo(remoteDON),
+					*toDONInfo(myDON),
+					w.dispatcher,
+					aggregator,
+					w.lggr,
+				)
+				return triggerCap, nil
+			}
+			err := w.addToRegistryAndSetDispatcher(ctx, capability, remoteDON, newTriggerFn)
+			if err != nil {
+				return fmt.Errorf("failed to add trigger shim: %w", err)
+			}
+		case capabilities.CapabilityTypeAction:
+			w.lggr.Warn("no remote client configured for capability type action, skipping configuration")
+		case capabilities.CapabilityTypeConsensus:
+			w.lggr.Warn("no remote client configured for capability type consensus, skipping configuration")
+		case capabilities.CapabilityTypeTarget:
+			newTargetFn := func(info capabilities.CapabilityInfo) (capabilityService, error) {
+				client := target.NewClient(
+					info,
+					*toDONInfo(myDON),
+					w.dispatcher,
+					defaultTargetRequestTimeout,
+					w.lggr,
+				)
+				return client, nil
+			}
+
+			err := w.addToRegistryAndSetDispatcher(ctx, capability, remoteDON, newTargetFn)
+			if err != nil {
+				return fmt.Errorf("failed to add target shim: %w", err)
+			}
+		default:
+			w.lggr.Warnf("unknown capability type, skipping configuration: %+v", capability)
+		}
+	}
+	return nil
+}
+
+type capabilityService interface {
+	capabilities.BaseCapability
+	remotetypes.Receiver
+	services.Service
+}
+
+func (w *launcher) addToRegistryAndSetDispatcher(ctx context.Context, capabilityInfo kcr.CapabilitiesRegistryCapabilityInfo, don kcr.CapabilitiesRegistryDONInfo, newCapFn func(info capabilities.CapabilityInfo) (capabilityService, error)) error {
+	fullCapID := fmt.Sprintf("%s@%s", capabilityInfo.LabelledName, capabilityInfo.Version)
+	info, err := capabilities.NewRemoteCapabilityInfo(
+		fullCapID,
+		toCapabilityType(capabilityInfo.CapabilityType),
+		fmt.Sprintf("Remote Capability for %s", fullCapID),
+		toDONInfo(don),
+	)
+	if err != nil {
+		return err
+	}
+	w.lggr.Debugw("Adding remote capability to registry", "id", info.ID, "don", info.DON)
+	capability, err := newCapFn(info)
+	if err != nil {
+		return err
+	}
+
+	err = w.registry.Add(ctx, capability)
+	if err != nil {
+		// If the capability already exists, then it's either local
+		// or we've handled this in a previous syncer iteration,
+		// let's skip and move on to other capabilities.
+		if errors.Is(err, ErrCapabilityAlreadyExists) {
+			return nil
+		}
+
+		return err
+	}
+
+	err = w.dispatcher.SetReceiver(
+		fullCapID,
+		fmt.Sprint(don.Id),
+		capability,
+	)
+	if err != nil {
+		return err
+	}
+	w.lggr.Debugw("Setting receiver for capability", "id", fullCapID, "donID", don.Id)
+	err = capability.Start(ctx)
+	if err != nil {
+		return err
+	}
+	w.subServices = append(w.subServices, capability)
+	return nil
+}
+
+var (
+	defaultTargetRequestTimeout = time.Minute
+)
+
```

Comment:
nit: 'enableExternalAccess' sort of implies it's a security thing, perhaps 'exposeCapabilities'